src/server/game/Maps/GridMap.cpp

```diff
@@ -656,8 +658,8 @@ ZLiquidStatus GridMap::GetLiquidStatus(float x, float y, float z, Optional<map_l
 
     // Get water level
     float liquid_level = _liquidMap ? _liquidMap[lx_int*_liquidWidth + ly_int] : _liquidLevel;
-    // Get ground level (sub 0.2 for fix some errors)
-    float ground_level = getHeight(x, y);
+    // Get ground level (sub 0.02 for fix some errors)
+    float ground_level = getHeight(x, y) - GROUND_LEVEL_OFFSET_HACK;
```

Comment:
Are you sure that this is the way to go? `ground_level` variable is returned from this function in `LiquidData`

Reply:
You mean that this might cause issues when comparing that level against the wmo level?

Reply:
I mean that if every function in the callstack starts to "slightly adjust" inputs/outputs instead of only checks, you end up accumulating them for even greater inaccuracy

Reply:
fair enough. I'll relocate the check to the check below and test again

Reply:
tested, no regressions. change has been made